src/app/services/transmart-services/transmart-resource.service.ts

```diff
@@ -279,6 +284,7 @@ export class TransmartResourceService {
       if (this.autosaveSubjectSets) {
         this.savePatientSet('temp', constraint)
           .subscribe((subjectSet: SubjectSet) => {
+            this._savedSubjectSetConstraints = constraint;
```

Comment:
use getters instead of the field itself

Reply:
also, updateInclusionExclusionCounts is only going to be invoked with the user clicks the update_1 button, I thought we try to save subject set even though the user has not clicked the update button

Reply:
that was the first idea, but as we discussed yesterday, saving subject set can take some time, when there is a lot of data and might not be a good performance improvement to save it before the export to use patient set constraint instead of direct constraint from step1_2. This is something I am not sure about, so maybe we should discuss it with others.